src/text_formatter/text_formatter.ts

```diff
@@ -0,0 +1,140 @@
+
+// --------------------------------------
+// @author: Ronja K.
+// ---------------------------------------
+
+// ---------format the input of a user into a full list of senteces including a list of words, formatted expressions, and paragraph marker-------
+export function textToListFormat(input: String) {
+
+	input = input.replace("$", "$EXPRESSIONMARKER");                            // prepare expressionsdetector; this is needed as marker to finde the expressions after the $ disappear by calling the split-method
+	input = input.replace("\n", "\n PARAGRAPHMARKER.");                         // prepare paragraphmarker
+
+	const separator = /[\n.!?$]+/g;                                              // List all the values the input will get splitted; \n will split paragraphs, $ will split expressions, the rest split sentences
+
+	const splittedText = input.split(separator);
+	const temp = [];
+	// let sentenceCount = 0;                                                  //  |optional
+	// let expressionCount = 0;                                                //  |optional
+	// let paragraphCount = 0;                                                 //  |optional
+	for (let index = 0; index < splittedText.length; index++) {
+		let element = splittedText[index].trim();                               // whitespaces between sentences are not needed
+		if (element != "") {
+
+			if (element.match(/(EXPRESSIONMARKER)/g)) {						   // DETECT AND ADD EXPRESSIONS TO LIST
+				// temp.push("Expression_" + (expressionCount + 1));           //  |optional
+				// expressionCount++;                                          //  |optional
+				element = element.replace("EXPRESSIONMARKER", "");
+				const formattedExpression = expressionFormatter(element);
+				temp.push(formattedExpression);
+			} else if (element.match(/(PARAGRAPHMARKER)/g)) {				   // DETECT AND ADD PARAGRAPH TO LIST
+				// temp.push("Paragraph_" + (paragraphCount + 1));             // |optional
+				// paragraphCount = 0;                                         // |optional
+				element = element.replace(/(PARAGRAPHMARKER)/g, "abs");
+				const ListWithparagraph = [];										// store paragraph marker in a own list
+				ListWithparagraph.push(element);
+				temp.push(ListWithparagraph);
+			} else {															// DETECT AND ADD WORDS TO LIST
+				// temp.push("Sentence_" + (sentenceCount + 1));                // |optional
+				// sentenceCount++;                                             // |optional
+				const ListOfWords = sentenceIntoWordList(element);
+				temp.push(ListOfWords);
+			}
+
+		}
+
+	}
+	const listFormat = temp;
+	return listFormat;
+}
+
+// ---------format the words from a sentence-string into a list of words-------
+export function sentenceIntoWordList(input: String) {
+	const separator = /[ ,$]+/; // List all the values where the content of a sentence will get splitted: whitespace[ ] and comma[,]
+	const splittedSentenceIntoListOfWords = input.split(separator);
```

Comment:
Instead of splitting, you could just scan for words: https://regex101.com/r/w47nEF/1